tools/adev-api-extraction/extract_api_to_json.bzl

```diff
@@ -0,0 +1,109 @@
+load("@build_bazel_rules_nodejs//:providers.bzl", "run_node")
+
+def _extract_api_to_json(ctx):
+    """Implementation of the extract_api_to_json rule"""
+
+    # Define arguments that will be passed to the underlying nodejs program.
+    args = ctx.actions.args()
+
+    # Use a param file because we may have a large number of inputs.
+    args.set_param_file_format("multiline")
+    args.use_param_file("%s", use_always = True)
+
+    # Pass the module_name for the extracted APIs. This will be something like "@angular/core".
+    args.add(ctx.attr.module_name)
+
+    # Pass the module_label for the extracted APIs, This is something like core for "@angular/core".
+    args.add(ctx.attr.module_label)
+
+    # Pass the set of private modules that should not be included in the API reference.
+    args.add_joined(ctx.attr.private_modules, join_with = ",")
+
+    # Pass the entry_point for from which to extract public symbols.
+    args.add(ctx.file.entry_point)
+
+    # Pass the set of source files from which API reference data will be extracted.
+    args.add_joined(ctx.files.srcs, join_with = ",")
+
+    # Pass the name of the output JSON file.
+    json_output = ctx.outputs.output_name
+    args.add(json_output.path)
+
+    # Pass the import path map
+    # TODO: consider module_mappings_aspect to deal with path mappings instead of manually
+    # specifying them
+    # https://github.com/bazelbuild/rules_nodejs/blob/5.x/internal/linker/link_node_modules.bzl#L236
```

Comment:
Nit: update this TODO. With our new Bazel toolchain, this whole concept no longer exists (probably `module_name` attributes still need to be cleaned up though).

This way seems correct, but even better would be just leveraging the real tsconfig from `src/bazel-tsconfig-build` with the "real path mappings".

Reply:
we can follow-up in the future. It would mostly be about reading the tsconfig from `src` simply. btw: I still see the TODO, but not a big deal